cmd/kubeops/internal/handler/handler_test.go

```diff
@@ -46,11 +47,22 @@ func newConfigFixture(t *testing.T) *Config {
 	}
 }
 
+var releaseComparer = cmp.Comparer(func(x release.Release, y release.Release) bool {
+	return x.Name == y.Name &&
+		x.Version == y.Version &&
+		x.Namespace == y.Namespace &&
+		x.Info.Status == y.Info.Status &&
+		x.Chart.Name() == y.Chart.Name() &&
+		x.Manifest == y.Manifest &&
+		cmp.Equal(x.Config, y.Config) &&
+		cmp.Equal(x.Hooks, y.Hooks)
```

Comment:
Maybe a bit weird indentation? I'd probably do something like this by the way:

```go
return and(bool[]{
	x.Name == y.Name,
	x.Version == y.Version,
	// ...
	cmp.Equal(x.Hooks, y.Hooks),
})
```

But it seems like I'd have to implement `and` myself:

```go
func and(xs []bool) bool {
	for _, x := range xs {
		if !x {
			return false
		}
	}
	return true
}
```

Reply:
Maybe as a Variadic Function?
```go
func and(exps ...bool) bool {
	for _, exp := range exps {
		if !exp {
			return false
		}
	}
	return true
}
```

Reply:
we would need to maintain extra code. IMO it's fine as it is (I don't see the indentation weird)

Reply:
That's neat, @latiif!

Reply:
> we would need to maintain extra code. IMO it's fine as it is (I don't see the indentation weird)

Why would `x.Name == y.Name &&` not be indented and the rest of the expression indented one level? What does indentation represent then? If one wants to avoid a separate `and` function, surely the below formatting is more logical?

```go
return (
	x.Name == y.Name &&
	x.Version == y.Version &&
	// ...
	cmp.Equal(x.Hooks, y.Hooks)
)
```

(I'm personally not very fond of it either, because I'd prefer that adding another line 1) be as easy as duplicating and modifying the last one, and 2) not create a three-line diff. But that's orthogonal to my indentation concern.)

Reply:
I actually just moved this code from the existing tests of @latiif 's . Happy to improve it, but don't see a problem as is either.

Anyway, I'll hold off updating and landing this PR until @latiif's test PRs have landed.

Reply:
I decided to summarize the three alternatives. Please let me know what you think – not least because I think it's an interesting matter in general (and don't forget my earlier question: "What does indentation represent then?"). :smiley: 

### `and` function
#### Pros
* Very easy to add or remove a condition at the beginning or end.
* Clear diff when adding or removing a condition at the beginning or end.
* DRY: Doesn't repeat `&&` several times. (Debatable, but I for one believe the intended conveyed meaning is "_a_, _b_ and _c_ ANDed together", not "_a_ ANDed with _b_ ANDed with _c_ …".)
* Indentation makes perfect sense.
#### Cons
* `and` must be maintained.

### `return (` with conditions on separate lines
#### Pros
* Very easy to add or remove a condition at the beginning.
* Clear diff when adding or removing a condition at the beginning.
* Indentation makes perfect sense.
#### Cons
* Not as easy to add or remove a condition at the end.
* Not very clear diff when adding or removing a condition at the end.
* WET: Repeats `&&` several times.

### `return x.Name == y.Name &&` (current PR content)
#### Pros
* Saves two lines compared to the other alternatives.
* ?
#### Cons
* Not as easy to add or remove a condition at the beginning or end.
* Not very clear diff when adding or removing a condition at the beginning or end.
* WET: Repeats `&&` several times.
* Indentation doesn't make sense. (Feel free to convince me of the contrary.)

Reply:
My issue with splitting up tests for different actions is that it would lead to duplication of code and maintaining a each test function on its own, meaning that we would end up with something similar to this:

```go
func TestGetAction(t *testing.T){
    //Defining the struct for the test scenarios
    type TestScenario struct{
        ⋮
        ⋮
    }
    //Filling of the test scenarios array
    tests := []TestScenario {
        ⋮
        ⋮
        ⋮
    }
    // Performing the actual test and checking of results
   for _, test := range tests {
		t.Run(test.Description, func(t *testing.T) {
	    	     ⋮
	    	     ⋮
		}
	}
}

func TestDeleteAction(t *testing.T){
    //Defining the struct for the test scenarios
    type TestScenario struct{
        ⋮
        ⋮
    }
    //Filling of the test scenarios array
    tests := []TestScenario {
        ⋮
        ⋮
        ⋮
    }
    // Performing the actual test and checking of results
   for _, test := range tests {
		t.Run(test.Description, func(t *testing.T) {
	    	        ⋮
	    	        ⋮
		}
	}
}
// Do the same for each new action
⋮
⋮
```
This IMHO leads to a bigger code base to maintain, my suggestion is to: 
1. Define the generic structure for test scenarios on a file-level (as opposed to having it inside the `TestXxxAction`.
2. Define a gerenic function that runs the tests extractin common code from the code snippet above (akin to `TestActions` but with an empty `tests=[]TestScenario{}`.
3. With each new action, define a function of the form `func getXxxActionTests() []TestScenario`, that returns tests for the `Xxx` action.
4. Populate the `tests` inside `TestActions` by calling a function that returns desired test scenarios. See below: 
```go
tests := []TestScenario {
    getGetActionTests(),
    getDeleteActionTests(),
    ⋮  
}
```
This would faciliate adding new test scenarios, grouping them and only requires us to maintain one function i.e. `TestActions`.

Reply:
> I decided to summarize the three alternatives. Please let me know what you think – not least because I think it's an interesting matter in general (and don't forget my earlier question: "What does indentation represent then?"). smiley

I didn't write the code, so when moving this function so that I could also use it in another test, I explicitly tried not to change Latiif's code (as little as possible, at least). You'll see this function was just copied and pasted in this PR, with its existing indentation.

That said, since it seems to be quite important to you, let me answer your questions with my own thoughts...

> What does indentation represent then?

I don't particularly like the indentation either (visually or semantically), but it's what the gofmt wants, so I don't usually bother thinking more about it. It is formatting after all and gofmt was created for this reason (ie. both to avoid spending too much time discussing formatting in PRs, as well as creating code which is formatted consistently across code bases etc.). So if you look at the actual source for go, you'll see similar formatting of long booleans throughout, just grep for "\&\&$" in the go source.
 
> ### `and` function
> #### Pros
> 
>     * Very easy to add or remove a condition at the beginning or end.
> 
>     * Clear diff when adding or removing a condition at the beginning or end.
> 
>     * DRY: Doesn't repeat `&&` several times. (Debatable, but I for one believe the intended conveyed meaning is "_a_, _b_ and _c_ ANDed together", not "_a_ ANDed with _b_ ANDed with _c_ …".)
> 
>     * Indentation makes perfect sense.
> 
> 
> #### Cons
> 
>     * `and` must be maintained.

I totally agree that using `and` results in the nicest formatting (visually, semantically and future diffs), though I don't agree that it's what we should do with our golang code, for the reasons above (which are also outlined in the original blog post at https://blog.golang.org/go-fmt-your-code too).

That said, even though I don't agree, I'm happy to update and use `and` here, in this specific case, for 3 reasons:
 1. You think it's important so I want to treat it as important,
 2. I do like the formatting more than what gofmt does (even though I don't think this is a good reason to change the code, I do feel the same niggle),
 3. I'm not keen to spend more time discussing the PR rather than landing the PR (and I don't like landing a PR, even with an approval, if a disagreement hasn't been resolved).

So, change done :)

Reply:
Thank you for a well-written reply with good points and important context, @absoludity!